### PR TITLE
TZUP-432: Add new columns for TB Monthly program configuration

### PR DESCRIPTION
**Pull request by sradziszewski**
JIRA task link: https://openlmis.atlassian.net/browse/TZUP-432

- - -
This change is [Reviewable](https://reviewable.io/reviews/OpenLMIS/openlmis-requisition/77)

### CODE REVIEW

File: src/main/java/org/openlmis/requisition/domain/requisition/RequisitionLineItem.java [hunk 1]
```diff
@@ -397,6 +433,13 @@ void updateFrom(RequisitionLineItem requisitionLineItem) {
 
       this.adjustedConsumption = requisitionLineItem.getAdjustedConsumption();
       this.averageConsumption = requisitionLineItem.getAverageConsumption();
+
```
Comment (pwargulak) [inline src/main/java/org/openlmis/requisition/domain/requisition/RequisitionLineItem.java:436]: Nitpicking: looks like unnecessary new line.

File: src/main/java/org/openlmis/requisition/domain/requisition/RequisitionLineItem.java [hunk 2]
```diff
@@ -333,6 +362,13 @@ static RequisitionLineItem newRequisitionLineItem(Importer importer,
     requisitionLineItem.setCalculatedOrderQuantityIsa(importer.getCalculatedOrderQuantityIsa());
     requisitionLineItem.setAdditionalQuantityRequired(importer.getAdditionalQuantityRequired());
```
Comment (pwargulak) [inline src/main/java/org/openlmis/requisition/domain/requisition/RequisitionLineItem.java:364]: Nitpicking: looks like unnecessary new line.

File: src/test/java/org/openlmis/requisition/domain/requisition/RequisitionLineItemDataBuilder.java [hunk 3]
```diff
@@ -68,6 +68,11 @@ public class RequisitionLineItemDataBuilder implements DataBuilder<RequisitionLi
   //this needs to be always idealStockAmount - stockOnHand
   private Integer calculatedOrderQuantityIsa = 50;
   private Integer additionalQuantityRequired = 0;
+  private Integer numberOfPatientsOnTreatmentNextMonth;
```
Comment (pwargulak) [inline src/test/java/org/openlmis/requisition/domain/requisition/RequisitionLineItemDataBuilder.java:71]: Why this doesn't have a default, but the others do?

**Review by pwargulak — CHANGES_REQUESTED** [review]
Left only some nitpicking comments. :)

**Review by kchelstowski — APPROVED** [review]
except one question and what @pwargulak has caught about a default value for one integer, LGTM

File: src/main/java/org/openlmis/requisition/domain/requisition/RequisitionLineItem.java (hunk 1 above)
Comment (sradziszewski) [reply src/main/java/org/openlmis/requisition/domain/requisition/RequisitionLineItem.java]: Empty line removed.

File: src/main/java/org/openlmis/requisition/domain/requisition/RequisitionLineItem.java (hunk 2 above)
Comment (sradziszewski) [reply src/main/java/org/openlmis/requisition/domain/requisition/RequisitionLineItem.java]: Empty line removed.

File: src/test/java/org/openlmis/requisition/domain/requisition/RequisitionLineItemDataBuilder.java (hunk 3 above)
Comment (sradziszewski) [reply src/test/java/org/openlmis/requisition/domain/requisition/RequisitionLineItemDataBuilder.java]: I added the value.

File: src/main/resources/db/migration/20230705133304288__add_columns_for_tb_monthly.sql [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+	INNER JOIN requisition.available_requisition_columns AS a ON a.name = 'quantityToIssue';
+
+INSERT INTO
+	requisition.columns_maps(requisitiontemplateid, requisitioncolumnid, definition, displayorder, indicator, isdisplayed, label, name, requisitioncolumnoptionid, source, key, tag)
+SELECT
+	t.id, a.id, a.definition, 12, a.indicator, false, a.label, a.name, NULL, 1, a.name, NULL
+FROM
+	requisition.requisition_templates AS t
+	INNER JOIN (SELECT requisitiontemplateid, count(*) FROM requisition.columns_maps GROUP BY requisitiontemplateid) AS c ON c.requisitiontemplateid = t.id
+	INNER JOIN requisition.available_requisition_columns AS a ON a.name = 'convertedQuantityToIssue';
```
[the remark this answers is not in the document]
Comment (sradziszewski) [reply src/main/resources/db/migration/20230705133304288__add_columns_for_tb_monthly.sql]: I added the empty line.

**Review by kchelstowski — APPROVED** [review]

**Review by pwargulak — APPROVED** [review]

**Pull request merged**